docs/_tooling/extensions/modularity/tests/test_modularity.py

```diff
@@ -0,0 +1,231 @@
+# *******************************************************************************
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+import pytest
+import json
+from types import SimpleNamespace
+from pathlib import Path
+from sphinx_needs.data import SphinxNeedsData
+from sphinx.testing.util import SphinxTestApp
+from docs._tooling.extensions.modularity import read_filter_tags
+
+
+#                    ╭──────────────────────────────────────────────────────────────────────────────╮
+#                    │                                  Unit Tests                                  │
+#                    ╰──────────────────────────────────────────────────────────────────────────────╯
+def test_read_filter_tags(sphinx_base_dir, caplog):
+    feature_flags_dir = sphinx_base_dir / "feature_flags"
+    feature_flags_dir.mkdir(exist_ok=True)
+    tag_str = "some-ip, feature1, feature2"
+    (feature_flags_dir / "filter_tags.txt").write_text(tag_str)
```

Comment:
Please extract such blocks to separate functions. I'd like to read the test to understand what it does. Without having to read how it does that.
This one sounds like write_tags_as_file(tags) or even better simulate_bazel_tags_file(tags)